api/src/test/java/com/epam/pipeline/manager/pipeline/FolderTemplateManagerTest.java

```diff
@@ -16,157 +16,216 @@
+        FolderTemplate childFolderTemplate = getFolderTemplate(dataStorageForChild, null,
+                childMetadata, childPermissionVO, TEST_NAME_2);
+        FolderTemplate folderTemplate = getFolderTemplate(dataStorageVO, childFolderTemplate,
+                metadata, permissionVO, TEST_NAME);
+
+        childFolder = getFolderChild(new S3bucketDataStorage(FOLDER_ID, DATASTORAGE_NAME_2, TEST_PATH_2));
+        folder = getFolder(new S3bucketDataStorage(FOLDER_CHILD_ID, DATASTORAGE_NAME, TEST_PATH));
+
+        doReturn(folder).when(mockFolderCrudManager)
+                .create(argThat(matches(Predicates.forFolderTemplate(folderTemplate.getName()))));
+        doReturn(childFolder).when(mockFolderCrudManager)
+                .create(argThat(matches(Predicates.forChildFolderTemplate(childFolderTemplate.getName()))));
+        doReturn(getMockSecurityEntity()).when(mockDataStorageManager)
+                .create(argThat(matches(Predicates.forDataStorage())), eq(true), eq(true), eq(false));
+        doReturn(getMockSecurityEntity()).when(mockDataStorageManager)
+                .create(argThat(matches(Predicates.forChildDataStorage())), eq(true), eq(true), eq(false));
+
+        //permissionGrantVOMapper doesn't work, so mapping it manually
+        doReturn(getPermissionGrantVOFrom(permissionVO, AclClass.FOLDER, FOLDER_ID)).when(mockPermissionGrantVOMapper)
+                .toPermissionGrantVO(eq(permissionVO));
+        doReturn(getPermissionGrantVOFrom(childPermissionVO, AclClass.FOLDER, FOLDER_CHILD_ID))
+                .when(mockPermissionGrantVOMapper).toPermissionGrantVO(eq(childPermissionVO));
+
+        doReturn(folder).when(mockEntityManager).load(eq(AclClass.FOLDER), eq(FOLDER_ID));
+        doReturn(childFolder).when(mockEntityManager).load(eq(AclClass.FOLDER), eq(FOLDER_CHILD_ID));
+        doReturn(new PrincipalSid(folder.getOwner())).when(aclService).createOrGetSid(
+                eq(permissionVO.getUserName().toUpperCase()), eq(permissionVO.getPrincipal()));
+        doReturn(new PrincipalSid(childFolder.getOwner())).when(aclService).createOrGetSid(
+                eq(childPermissionVO.getUserName().toUpperCase()), eq(childPermissionVO.getPrincipal()));
+
+        initAclEntity(folder, AclPermission.READ);
+        initAclEntity(childFolder, AclPermission.READ);
 
+        folderTemplateManager.createFolderFromTemplate(folder, folderTemplate);
 
-    @Before
-    public void setUp() {
-        doReturn(new MockS3Helper()).when(storageProviderManager).getS3Helper(any(S3bucketDataStorage.class));
+        verify(mockFolderCrudManager).create(argThat(matches(folder -> TEST_NAME.equals(folder.getName()))));
+        verify(mockFolderCrudManager).create(argThat(matches(folder -> TEST_NAME_2.equals(folder.getName()))));
+        verify(mockDataStorageManager).create(eq(dataStorageVO), eq(true), eq(true), eq(false));
+        verify(mockDataStorageManager).create(eq(dataStorageForChild), eq(true), eq(true), eq(false));
```

Comment:
You could use already defined predicates from `Predicates` for these calls.

Reply:
Fixed

Reply:
Same for data storage calls.

Reply:
Done